controllers/kuadrant_controller.go

```diff
@@ -163,13 +161,13 @@ func (r *KuadrantReconciler) Reconcile(eventCtx context.Context, req ctrl.Reques
 	return ctrl.Result{}, nil
 }
 
-func (r *KuadrantReconciler) unregisterExternalAuthorizer(ctx context.Context) error {
+func (r *KuadrantReconciler) unregisterExternalAuthorizer(ctx context.Context, kObj *kuadrantv1beta1.Kuadrant) error {
 	logger, _ := logr.FromContext(ctx)
 
-	err := r.unregisterExternalAuthorizerIstio(ctx)
+	isIstioInstalled, err := r.unregisterExternalAuthorizerIstio(ctx, kObj)
```

Comment:
Maybe in a next iteration `unregisterExternalAuthorizerIstio` and `unregisterExternalAuthorizerOSSM` can become one, and similarly `registerExternalAuthorizerIstio` and `registerExternalAuthorizerOSSM`. Fundamentally, and given the new abstractions now introduced with this PR, these pairs of functions are not very different between them – if any different at all.

`getIstioConfigObjects` could become simply `getConfigObjects` (or perhaps `getGatewayProviderConfigObjects` to be more literal with the name), and return all gateway provider config objects that need to be patched.

Istio or OSSM, it doesn't really matter. In the end, it all reduces to iterating over all configs, passing them into the calls to `common.UnregisterKuadrantAuthorizer` and `common.RegisterKuadrantAuthorizer`. Thanks to this PR, the details of how to patch each kind of config is decoupled from the caller, but abstracted by each config wrapper itself.

Reply:
Makes total sense now, I'll keep in mind this for the next refactor, that most possibly will be the plugin way of using GWs